concordium-contracts-common/src/types.rs

```diff
@@ -78,6 +78,9 @@ impl fmt::Display for AmountParseError {
     }
 }
 
+#[cfg(feature = "std")]
+impl std::error::Error for AmountParseError {}
```

Comment:
So this is a bit wonky. It uses the default implementation which will have `.description` return "Deprecated, use display".

Is this intended?

Reply:
That is the current intended way of implementing this, and seems to also be how the `thiserror::Error` derive macro behaves.